### ISSUE TITLE
Support option of delegating to multiple surefire providers 

### ISSUE THREAD

**Issue by MatousJobanek** [arquillian/smart-testing#282]
##### Issue Overview

In the issue #278 , there was introduced an option of delegating test execution to custom provider specified in Surefire's dependencies. But this works only for one provider. If there are more providers specified, ST picks only the first one and delegates the execution to it. 
The task for this issue is to delegate the classpath scan and the test execution to multiple surefire providers (if they are present in the list of Surefire's dependencies)

**Comment by bartoszmajsak** [comment]
Can this be closed @MatousJobanek?

**Comment by MatousJobanek** [comment]
no, this is another issue - different use case. Currently, it supports only one custom provider. This task is to support multiple providers.

**Comment by bartoszmajsak** [comment]
Ok, reference in the #281 got me confused
